syntax = "proto3";
package users;

message UserIdRequest {int32 user_id = 1;}
message BoolResponse   {bool ok = 1;}
message UserFull {
  int32  user_id = 1;
  string first_name = 2;
  string last_name = 3;
  string role = 4;
  string phone = 5;
  string avatar = 6;
  string passport_number = 7;
  string birth_date = 8;
  string tax_id = 9;
}

service UserService {
  rpc GetUser   (UserIdRequest) returns (UserFull);
  rpc BanUser   (UserIdRequest) returns (BoolResponse);
  rpc IsBanned  (UserIdRequest) returns (BoolResponse);
}
